### PR TITLE
fix(ci): make it work + publish with provenances

### PR DESCRIPTION
**Pull request by vladfrangu**
Fixes the flow for publishing/version bumping

Also adds in provenances, aka this shiny badge on npm

Google Chrome Beta - 2024-11-07 at 17 22 14@2x

### CODE REVIEW

File: .github/workflows/pre_release.yaml [hunk 1]
```diff
@@ -1,133 +1,141 @@
...
+
+      - name: Update package version in package.json
+        run: yarn version ${{ needs.release_metadata.outputs.version_number }}
+
+      - name: Update CHANGELOG.md
+        uses: DamianReeves/write-file-action@master
+        with:
+          path: CHANGELOG.md
+          write-mode: overwrite
+          contents: ${{ needs.release_metadata.outputs.changelog }}
```
Comment (janbuchar) [inline .github/workflows/pre_release.yaml:83]: Please add a "Update README" step somewhere around here

**Review by janbuchar — CHANGES_REQUESTED** [review]
Thanks, I just see one issue.

File: .github/workflows/pre_release.yaml [hunk 2]
```diff
@@ -1,133 +1,141 @@
...
+      - name: Build module
+        run: yarn build
+
+      - name: Pack with yarn
+        run: yarn pack
+
+      - name: Publish to NPM
+        run: |
+          # `yarn npm publish` does not currently support --provenance: https://github.com/yarnpkg/berry/issues/5430
+          npm publish package.tgz --provenance --access public
```
Comment (janbuchar) [inline .github/workflows/pre_release.yaml:139]: Is `--tag beta` missing here?

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -1,205 +1,207 @@
...
+      - name: Build module
+        run: yarn build
+
+      - name: Pack with yarn
+        run: yarn pack
+
+      - name: Publish to NPM
+        run: |
+          # `yarn npm publish` does not currently support --provenance: https://github.com/yarnpkg/berry/issues/5430
+          npm publish package.tgz --provenance --access public
```
Comment (janbuchar) [inline .github/workflows/release.yaml:155]: Is `--tag latest` missing here?

File: .github/workflows/release.yaml [hunk 4]
```diff
@@ -1,205 +1,207 @@
...
+          corepack enable
+          corepack prepare yarn@stable --activate
+
+      - name: Activate cache for Node.js 22
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+
+      - name: Update package version in package.json
+        run: yarn version ${{ needs.release_metadata.outputs.version_number }}
```
Comment (janbuchar) [inline .github/workflows/release.yaml:86]: Is this really equivalent to `--no-git-tag-version --allow-same-version` for `npm version`?

File: .github/workflows/pre_release.yaml [hunk 5]
```diff
@@ -1,133 +1,141 @@
...
+        uses: actions/setup-node@v4
+        with:
+          cache: yarn
+
+      - name: Update README
+        run: |
+          yarn pack
+
+      - name: Update package version in package.json
+        run: yarn version ${{ needs.release_metadata.outputs.version_number }}
```
Comment (janbuchar) [inline .github/workflows/pre_release.yaml:76]: Is this really equivalent to `--no-git-tag-version --allow-same-version` for `npm version`?

File: .github/workflows/release.yaml (hunk 3 above)
Comment (B4nan) [reply .github/workflows/release.yaml:155]: `--tag latest` is the default

File: .github/workflows/pre_release.yaml (hunk 1 above)
Comment (vladfrangu) [reply .github/workflows/pre_release.yaml:83]: We don't actually want to update the readme file as then we'd have beta readmes on prod website 😅

Comment (janbuchar) [reply .github/workflows/pre_release.yaml]: Right!

**Review by janbuchar — APPROVED** [review]
LGTM

**Pull request merged**